cmd/k8s-operator/dnsrecords.go

```diff
@@ -297,6 +303,11 @@ func (dnsRR *dnsRecordsReconciler) removeHeadlessSvcFinalizer(ctx context.Contex
 // proxy types, or any other Services, but it just returns an empty string if
 // that happens.
 func (dnsRR *dnsRecordsReconciler) fqdnForDNSRecord(ctx context.Context, headlessSvc *corev1.Service, logger *zap.SugaredLogger) (string, error) {
+	// For ProxyGroup egress, the FQDN is stored directly in the magic-dnsname annotation
+	if dnsRR.isProxyGroupEgressService(headlessSvc) {
```

Comment:
I might be incorrect, but isn't this annotation set on line [157](https://github.com/tailscale/tailscale/pull/16181/files#diff-5faf4c4be362426224bda410f2bd5dcf3e3118c20bcac04f05d86bd0d2dbbb37R157)? If so, this function is invoked on line [129](https://github.com/tailscale/tailscale/pull/16181/files#diff-5faf4c4be362426224bda410f2bd5dcf3e3118c20bcac04f05d86bd0d2dbbb37R129) and kills the reconcile if the returned string is empty. Surely at this stage the `annotationTSMagicDNSName` annotation won't have been set?

The reasonable explanation here is that this annotation is user managed? Maybe that is the case, but seeing as we're manipulating it in the reconcile that suggests it is not?
As